vulture_os/toolkit/api_parser/catonetworks/catonetworks.py

```diff
@@ -0,0 +1,156 @@
+logger = logging.getLogger('api_parser')
+
+
+class CatonetworksAPIError(Exception):
+    pass
+
+
+class CatonetworksParser(ApiParser):
+
+    URL = "https://api.catonetworks.com/api/v1/graphql2"
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.catonetworks_api_key = data["catonetworks_api_key"]
+        self.catonetworks_account_id = data["catonetworks_account_id"]
+
+        self.marker = data.get("catonetworks_marker", "")
+        self.event_filter_string = ""
+        self.event_subfilter_string = ""
+
+    def execute_query(self, query):
+        retry_count = 0
+        data = {'query':query}
+        headers = { 'x-api-key': self.catonetworks_api_key,'Content-Type':'application/json'}
+        while True:
+            if retry_count > 5:
+                logger.error(f"[{__parser__}]:execute_query: FATAL ERROR retry count exceeded", extra={'frontend': str(self.frontend)})
+                raise CatonetworksAPIError("FATAL ERROR retry count exceeded")
+            try:
+                response = requests.post(
+                    url=self.URL,
+                    data=json.dumps(data).encode("ascii"),
+                    headers=headers,
+                    proxies=self.proxies,
+                    verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl,
+                    timeout=30
+                )
+            except Exception as e:
+                logger.info(f"[{__parser__}]:execute_query: {e}", extra={'frontend': str(self.frontend)})
+                time.sleep(20)
+                self.update_lock()
+                retry_count += 1
+                continue
+            break
+        response.raise_for_status()
+        result = response.json()
+        if "errors" in result:
+            logger.error(f"[{__parser__}]:execute_query: Error while getting logs. reason : {response.content}", extra={'frontend': str(self.frontend)})
+            raise CatonetworksAPIError(f"Error while getting logs. reason : {response.content}")
+        return result
+
+    def get_logs(self):
+        query = '''{
+            eventsFeed(
+                accountIDs: [''' + self.catonetworks_account_id + ''']
+                marker: "''' + self.marker + '''"
+                filters: [''' + self.event_filter_string + "," + self.event_subfilter_string + ''']
+            ) {
+                marker
+                fetchedCount
+                accounts {
+                    id
+                    errorString
+                    records {
+                        time
+                        fieldsMap
+                    }
+                }
+            }
+        }'''
+        result = self.execute_query(query)
+        self.marker = result["data"]["eventsFeed"]["marker"]
+        self.fetched_count = int(result["data"]["eventsFeed"]["fetchedCount"])
+        return result["data"]["eventsFeed"]["accounts"][0]["records"]
```

Comment:
not a fan of using a class attribute for something you sxhould have clearly passed as a return to the function